src/Cooler.sol

```diff
@@ -237,9 +239,13 @@ contract Cooler is Clone {
     ) external returns (uint256 loanID) {
         Request memory req = requests[reqID_];
 
-        // If necessary, ensure lender implements the CoolerCallback abstract.
-        if (isCallback_ && !CoolerCallback(msg.sender).isCoolerCallback()) revert NotCoolerCallback();
+        // Loan callbacks are only allowed if:
+        //  1. The loan request has been created via a trusted lender.
+        //  2. The lender signals that it implements the CoolerCallback Abstract.
+        bool callback = (isCallback_ && msg.sender == req.requester);
```

Comment:
What this means is that the entity that provides the loan must also put up the collateral.

Which really means that it must be a contract that both takes collateral from the borrower and does the loan.

Which is not that far from saying that only the Clearinghouse can use a Cooler.

If that's the intention, then there are much simpler designs for lending. This would mean the orderbook mechanism of the Cooler is never used.

Overall, this fix seems contrary to the design of the Cooler.

My recommended fix is that reverting in onRepay or onDefault not cause the overall transaction to revert.